firebase/functions/hotline.js

```diff
@@ -0,0 +1,171 @@
+  // what we assume our call center hours
+  // in germany are
+  const hour = now.getHours();
+
+  let partOfDay;
+  switch (true) {
+    case hour >= 0 && hour < 10:
+      partOfDay = 'Night';
+      break;
+    case hour >= 10 && hour < 14:
+      partOfDay = 'Morning';
+      break;
+    case hour >= 14 && hour < 18:
+      partOfDay = 'Afternoon';
+      break;
+    case hour >= 18:
+      partOfDay = 'Night';
+      break;
+    default:
+      throw new Error(`Invalid hour ${hour}`);
+  }
+
+  return `${day} ${partOfDay}`;
+}
+
+async function getAuthorizedClient() {
+  const snapshot = await admin.database().ref(DB_TOKEN_PATH).once('value');
+  const oauthTokens = snapshot.val();
+  oauthClient.setCredentials(oauthTokens);
+
+  return oauthClient;
+}
+
+async function getValues(params) {
+
+  const sheets = google.sheets({ version: 'v4', auth: await getAuthorizedClient() });
+
+  return new Promise((resolve, reject) => {
+    sheets.spreadsheets.values.get(params, (err, res) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+}
+
+function configureGoogleApiAccess(req, res) {
+  res.set('Cache-Control', 'private, max-age=0, s-maxage=0');
+  res.redirect(oauthClient.generateAuthUrl({
+    access_type: 'offline',
+    scope: SCOPES,
+    prompt: 'consent',
+  }));
+}
+
+async function oauthCallback(req, res) {
+  res.set('Cache-Control', 'private, max-age=0, s-maxage=0');
+  const { code } = req.query;
+  try {
+    const { tokens } = await oauthClient.getToken(code);
+    await admin.database().ref(DB_TOKEN_PATH).set(tokens);
+    return res.send('Successfully connected google sheets with firebase functions').status(200);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.log(err);
+    return res.status(400).send(err);
+  }
+}
+
+async function getPeopleForShift(shift) {
+  const params = {
+    spreadsheetId: SPREADSHEET_ID,
+    range: RANGE,
+  };
+
+  const res = await getValues(params);
+  const rows = res.data.values;
+
+  if (!rows.length) {
+    throw new Error('No data found');
+  }
+
+  return rows
+    .map((row) => {
+      const number = row[1];
+      const availability = row.slice(2, row.length);
+
+      const rowIdx = weekdayIndex[shift];
+
+      return availability[rowIdx] === 'ja' ? number : null;
+    })
+    .filter(Boolean);
+}
+
+async function handleIncomingCall(req, res) {
+  // TODO: Secure with some kind of token as this exposes our phone numbers!
+  const shift = getCurrentShift();
+
+  try {
+    const people = await getPeopleForShift(shift);
+    if (people.length === 0) {
+      // Make it easier for Twilio Flow to check for empty string
+      res.status(200).send('');
+    } else {
+      res.status(200).send(people.join(','));
+    }
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(`An error occurred while retrieving numbers: ${err}`);
+    res.send('').status(500).end();
```

Comment:
I think you have to set the status before you call send as it will terminate the response. Furthermore, I think you don't need .end() if you use .send()
Anyways:
```suggestion
    res.status(500).end();
```
Should send the error response without any data.